custom_components/tahoma/coordinator.py

```diff
@@ -105,6 +105,22 @@ async def _async_update_data(self) -> Dict[str, Device]:
                 event.new_state,
             )
 
+            if event.failure_type_code:
```

Comment:
Wondering if cannot make this generics, instead of limiting this to the failure.